core/trino-spi/src/main/java/io/trino/spi/ptf/TableFunctionAnalysis.java

```diff
@@ -36,12 +45,15 @@
 public final class TableFunctionAnalysis
 {
     private final Optional<Descriptor> returnedType;
+    private final Map<String, Set<String>> requiredColumns;
```

Comment:
What's the key and the values this map? Add that to a comment.

Reply:
This is a map from table argument name to set of column names requested from that table argument. I added a comment.